Makefile

```diff
@@ -14,6 +14,32 @@ all_to_check=$(files_to_check) $(dirs_to_check)
 .PHONY: ci
 ci: deps checkgofmt vet staticcheck ineffassign predeclared test
 
+BIN_DIR := $(GOPATH)/bin
```

Comment:
I don't understand the need for the rest of the changes in this file.

The directory you are in should only matter if you do _not_ have the `GO111MODULE` env var set. (The way it decides whether to use modules or not, by default, is if you invoke it from a sub-directory of `GOPATH` IIRC.)